JustSaying.Sample.Restaurant.KitchenConsole/OrderPlacedEventHandler.cs

```diff
@@ -0,0 +1,49 @@
+using System.Threading.Tasks;
+using JustSaying.Messaging;
+using JustSaying.Messaging.MessageHandling;
+using JustSaying.Sample.Restaurant.Models;
+using Microsoft.Extensions.Logging;
+
+namespace JustSaying.Sample.Restaurant.KitchenConsole
+{
+    public class OrderPlacedEventHandler : IHandlerAsync<OrderPlacedEvent>
+    {
+        private readonly IMessagePublisher _publisher;
+        private readonly ILogger<OrderPlacedEventHandler> _log;
+
+        /// <summary>
+        /// Handles messages of type OrderPlacedEvent
+        /// Takes a dependency on IMessagePublisher so that further messages can be published 
+        /// </summary>
+        public OrderPlacedEventHandler(IMessagePublisher publisher, ILogger<OrderPlacedEventHandler> log)
+        {
+            _publisher = publisher;
+            _log = log;
+        }
+
+        public async Task<bool> Handle(OrderPlacedEvent message)
+        {
+            _log.LogInformation("Order {orderId} for {description} received", message.OrderId, message.Description);
+
+            // TODO: handle the actual order
+
+            _log.LogInformation("Order {orderId} ready", message.OrderId);
+
+            var orderReadyEvent = new OrderReadyEvent
```

Comment:
Wrap the "logic" in a try-catch, logging the exception in the catch block and returning false, e.g.:

```csharp
catch (Exception ex)
{
    _logger.LogError(ex, "Failed to handle message for {orderId}", message.OrderId);
    return false;
}
````

Reply:
fixed

Reply:
I rebased on master and get:
```
error CA1031: Modify 'Handle' to catch a more specific exception type, or rethrow the exception.
```

I've suppressed it via
```xml
<NoWarn>.............;CA1031</NoWarn>
```